lib/asm-parser.js

```diff
@@ -433,18 +494,24 @@ class AsmParser extends AsmRegex {
             if (match) {
                 func = match[2];
                 if (this.isUserFunction(func)) {
-                    result.push({text: func + ":", source: null});
+                    astResult.push({
+                        text: func + ":",
+                        source: null,
+                        labels: labelsInLine
+                    });
+                    labelDefinitions[func] = lineNumber;
+                    ++lineNumber;
```

Comment:
Is the line number meant to increase on every line? Or just on every `astResult` ?

If so probably the safest thing is not the `(line, lineNumber0Indexed)` I mentioned above, but to use the `astResult.length` as the line number: so again it can't get out of step with the actual line number, and ther'es no possible accounting issues :) That would work in the earlier case too (in the non-binary asm parser).

Reply:
Good tip, thanks for this :)